site/pages/dashboard.page.tsx

```diff
@@ -29,8 +30,27 @@ export interface DashboardProps {
     newDisruptionId: string;
 }
 
-const mapDisruptions = (disruptions: Disruption[]) =>
-    sortDisruptionsByStartDate(disruptions).map((disruption) => {
+const mapDisruptions = (disruptions: Disruption[]) => {
+    return sortDisruptionsByStartDate(disruptions).map((disruption) => {
+        let maxEndDate: Dayjs | null = getDate().subtract(100, "years");
+
+        disruption.validity?.forEach((validity) => {
+            const repeatsEndDate =
+                validity.disruptionRepeats === "daily" && validity.disruptionDailyRepeatsEndDate
+                    ? getFormattedDate(validity.disruptionDailyRepeatsEndDate)
+                    : validity.disruptionRepeats === "weekly" && validity.disruptionWeeklyRepeatsEndDate
+                    ? getFormattedDate(validity.disruptionWeeklyRepeatsEndDate)
+                    : validity.disruptionEndDate && validity.disruptionEndTime
+                    ? getDatetimeFromDateAndTime(validity.disruptionEndDate, validity.disruptionEndTime)
+                    : null;
```

Comment:
I think this code needs to be doing the following: 

- map over the validity periods to see if any of them contain 'disruptionRepeats'.
- If any of them repeat, check to see if they have an end date.
- If they dont, give them a max end date (the function to get a max end date should live in the Dates.ts file - we shouldnt be doing the subtraction here).

Then, the page already gets the final end date of the final validity period and uses it as the end date to show on the screen.

Reply:
The `sortDisruptionsByStartDate` method would sort the validity records as per the start date. This is how the earliest start date is obtained for the disruption.

The end date is obtained from the last validity record. This record was sorted based on start date and has not accounted for ending on date. 

If there are records with ending on date later than the end date of the last record this data is not captured because its not the last validity record. As the start date of this record with ending on date could be before the start date of the record in the last validity index

Reply:
But validity periods cant overlap, which means the end date of the last one in the array must be the latest end date?